File: aggregate-calc-backend/app/schemas/university.py
from datetime import datetime
from enum import Enum
from pydantic import BaseModel, ConfigDict

class ScreeningType(str, Enum):
    POST_UTME = "POST_UTME"
    NONE = "NONE"
    NO_OLEVEL_POINTS = "NO_OLEVEL_POINTS"

class UniversityUpdate(BaseModel):
    # name: str
    # short_name: str
    # state: str
    # ownership: str | None = None
    # website: str | None = None
    logo_url: str | None = None
    screening_type: ScreeningType
    is_active: bool

class UniversityCreate(BaseModel):
    name: str
    short_name: str
    state: str
    ownership: str | None = None
    website: str | None = None
    logo_url: str | None = None
    screening_type: ScreeningType
    is_active: bool = True

class UniversityResponse(BaseModel):
    id: int
    name: str
    short_name: str
    state: str
    ownership: str | None
    website: str | None
    logo_url: str | None
    screening_type: ScreeningType
    is_active: bool
    # created_at: datetime
    # updated_at: datetime

    model_config = ConfigDict(from_attributes=True)

File: aggregate-calc-frontend/src/types/university.ts
// Replaces src/types/university.ts — brought in line with the actual
// `universities` table. Confirm the `ownership` string values match what's
// really stored (assumed "Federal" | "State" | "Private" below).

export type ScreeningType = "POST_UTME" | "NONE" | "NO_OLEVEL_POINTS";
export type Ownership = "Federal" | "State" | "Private";

export interface University {
  id: number;
  name: string;
  short_name: string;
  state: string;
  logo_url: string | null;
  screening_type: ScreeningType;
  ownership: Ownership;
  website: string | null;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}